src/Raven.Server/Web/System/BackupDatabaseHandler.cs

```diff
@@ -91,6 +95,227 @@ await using (var writer = new AsyncBlittableJsonTextWriter(context, ResponseBody
             }
         }
 
+        [RavenAction("/backup-history-details", "GET", AuthorizationStatus.ValidUser, EndpointType.Read)]
+        public async Task GetBackupHistoryDetails()
+        {
+            var name = GetQueryStringValueAndAssertIfSingleAndNotEmpty("database");
+            var taskId = GetLongQueryString("taskId");
+            var createdAtTicksAsId = GetLongQueryString("id");
+            var nodeTag = GetStringValuesQueryString("nodeTag");
+
+            var currentNodeTag = ServerStore.NodeTag;
+            if (nodeTag != currentNodeTag)
+                throw new ArgumentException("wrong node"); //todo: rewrite message or entire functionality
+            var documentDatabase = await ServerStore.DatabasesLandlord.TryGetOrCreateResourceStore(name);
+
+            using (ServerStore.ContextPool.AllocateOperationContext(out TransactionOperationContext context))
+            using (context.OpenReadTransaction())
+            await using (var writer = new AsyncBlittableJsonTextWriter(context, ResponseBodyStream()))
+            {
+                var key = $"values/{name}/backup-history/{BackupHistoryItemType.Details}/{taskId}/{nodeTag}/{createdAtTicksAsId}";
+
+                writer.WriteStartObject();
+                writer.WritePropertyName(nameof(BackupHistoryItemType.Details));
+
+                using (documentDatabase.ConfigurationStorage.BackupHistoryStorage.ReadBackupHistoryDetails(key, out var details))
+                {
+                    writer.WriteObject(details);
+                }
+
+                writer.WriteEndObject();
+            }
+        }
+
+        [RavenAction("/backup-history", "GET", AuthorizationStatus.ValidUser, EndpointType.Read)]
+        public async Task GetBackupHistory()
+        {
+            var name = GetQueryStringValueAndAssertIfSingleAndNotEmpty("database");
+            var taskIdFromRequest = GetLongQueryString("taskId", required: false);
+            var nodeTagFromRequest = GetStringValuesQueryString("nodeTag", required: false);
+
+            var currentNodeTag = ServerStore.NodeTag;
+            var documentDatabase = await ServerStore.DatabasesLandlord.TryGetOrCreateResourceStore(name);
```

Comment:
This looks wrong. 
If you need to access a specific database the EP should be `/databases/*/...` and inherit from the `DatabaseRequestHandler`

Reply:
I would expect this EP to fetch a cluster-wide info

Reply:
Created a separate Handler for BackupHistory.